cmd/kubectl-gadget/deploy.go

```diff
@@ -580,36 +607,57 @@ func runDeploy(cmd *cobra.Command, args []string) error {
 		daemonSet, handlingDaemonSet := object.(*appsv1.DaemonSet)
 		if handlingDaemonSet {
 			daemonSet.Spec.Template.Annotations["inspektor-gadget.kinvolk.io/option-hook-mode"] = hookMode
-			daemonSet.Spec.Template.Annotations["container.apparmor.security.beta.kubernetes.io/gadget"] = appArmorprofile
+
 			daemonSet.Namespace = gadgetNamespace
 
 			// Inspektor Gadget used to require hostPID=true. This is no longer
 			// required, so keep hostPID=false unless the user explicitly
 			// requests it for compatibility with older clusters.
 			daemonSet.Spec.Template.Spec.HostPID = legacyHostPID
 
+			if seccompProfile != "" {
+				path := "operator/gadget/profile.json"
+				daemonSet.Spec.Template.Spec.SecurityContext = &v1.PodSecurityContext{
+					SeccompProfile: &v1.SeccompProfile{
+						Type:             v1.SeccompProfileTypeLocalhost,
+						LocalhostProfile: &path,
+					},
+				}
+			}
+
 			if !printOnly {
-				// The "kubernetes.io/os" node label was introduced in v1.14.0
-				// (https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.14.md.)
-				// Remove this if the cluster is older than that to allow Inspektor Gadget to work there.
 				serverInfo, err := discoveryClient.ServerVersion()
 				if err != nil {
 					return fmt.Errorf("getting server version: %w", err)
 				}
 
 				serverVersion := k8sversion.MustParseSemantic(serverInfo.String())
+
+				// The "kubernetes.io/os" node label was introduced in v1.14.0
+				// (https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG/CHANGELOG-1.14.md.)
+				// Remove this if the cluster is older than that to allow Inspektor Gadget to work there.
 				if serverVersion.LessThan(k8sversion.MustParseSemantic("v1.14.0")) {
 					delete(daemonSet.Spec.Template.Spec.NodeSelector, "kubernetes.io/os")
 				}
-			}
 
-			if seccompProfile != "" {
-				path := "operator/gadget/profile.json"
-				daemonSet.Spec.Template.Spec.SecurityContext = &v1.PodSecurityContext{
-					SeccompProfile: &v1.SeccompProfile{
-						Type:             v1.SeccompProfileTypeLocalhost,
-						LocalhostProfile: &path,
-					},
+				// Before 1.30, AppArmor profile was set as annotation, but since 1.30
+				// it has specific types:
+				// https://kubernetes.io/docs/tutorials/security/apparmor/#securing-a-pod
+				if serverVersion.AtLeast(k8sversion.MustParseSemantic("v1.30.0")) {
+					delete(daemonSet.Spec.Template.Annotations, "container.apparmor.security.beta.kubernetes.io/gadget")
```

Comment:
What about updating the manifests so they work right away with k8s > 1.30?

Reply:
Good point, I was thinking old versions of k8s will ignore this field but that's not the case.

Reply:
btw, currently it's not possible to apply the manifest directly, see #3389